samples/general/mobile_nerf_rayquery/mobile_nerf_rayquery.cpp

```diff
@@ -0,0 +1,1395 @@
+
+	std::vector<VkDynamicState> dynamic_state_enables = {
+	    VK_DYNAMIC_STATE_VIEWPORT,
+	    VK_DYNAMIC_STATE_SCISSOR};
+	VkPipelineDynamicStateCreateInfo dynamic_state =
+	    vkb::initializers::pipeline_dynamic_state_create_info(
+	        dynamic_state_enables.data(),
+	        static_cast<uint32_t>(dynamic_state_enables.size()),
+	        0);
+
+	VkPipelineMultisampleStateCreateInfo multisample_state = vkb::initializers::pipeline_multisample_state_create_info(VK_SAMPLE_COUNT_1_BIT, 0);
+
+	// No need for Vertex bindings and attributes
+	VkPipelineVertexInputStateCreateInfo vertex_input_state{};
+	vertex_input_state.sType = VK_STRUCTURE_TYPE_PIPELINE_VERTEX_INPUT_STATE_CREATE_INFO;
+
+	VkGraphicsPipelineCreateInfo pipeline_create_info = vkb::initializers::pipeline_create_info(pipeline_layout, render_pass_nerf, 0);
+	pipeline_create_info.pVertexInputState            = &vertex_input_state;
+	pipeline_create_info.pInputAssemblyState          = &input_assembly_state;
+	pipeline_create_info.pRasterizationState          = &rasterization_state;
+	pipeline_create_info.pColorBlendState             = &color_blend_state;
+	pipeline_create_info.pMultisampleState            = &multisample_state;
+	pipeline_create_info.pViewportState               = &viewport_state;
+	pipeline_create_info.pDepthStencilState           = &depth_stencil_state;
+	pipeline_create_info.pDynamicState                = &dynamic_state;
+	pipeline_create_info.subpass                      = 0;
+	pipeline_create_info.stageCount                   = static_cast<uint32_t>(shader_stages.size());
+	pipeline_create_info.pStages                      = shader_stages.data();
+
+	// Only need one pipeline in rayquery
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipeline));
+}
+
+void MobileNerfRayQuery::build_command_buffers()
+{
+	if (use_native_screen_size)
+	{
+		view_port_height = height;
+		view_port_width  = width;
+	}
+
+	// In case the screen is resized, need to update the storage image size and descriptor set
+	// Note that the texture_rendered image has already been recreated at this point
+	if (!prepared)
+	{
+		setup_framebuffers();
+	}
+
+	VkCommandBufferBeginInfo  command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+	std::vector<VkClearValue> clear_values;
+
+	clear_values.resize(2);
+	clear_values[0].depthStencil = {1.0f, 0};
+	clear_values[1].color        = {{1.0f, 1.0f, 1.0f, 1.0f}};
+
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderPass               = render_pass_nerf;
+	render_pass_begin_info.renderArea.offset.x      = 0;
+	render_pass_begin_info.renderArea.offset.y      = 0;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = clear_values.size();
```

Comment:
`static_cast<uint32_t>` missing?

Reply:
Added this and a few more that were missing. Thanks.